windows/RNCamera/RCTCameraModule.cs

```diff
@@ -90,35 +87,58 @@ public override string Name
         public CameraForViewManager CameraManager { get; } = new CameraForViewManager();
 
         [ReactMethod]
-        public async void capture(JObject options, IPromise promise)
+        public async void checkMediaCapturePermission(IPromise promise)
         {
-            var viewTag = options.Value<int>("view");
+            MediaCapture newCapture = new MediaCapture();
+            try
+            {
+                await newCapture.InitializeAsync();
+            }
+            catch
+            {
+                promise.Resolve(false);
+                return;
+            }
+
+            promise.Resolve(true);
+        }
+
+        [ReactMethod]
+        public async void takePicture(JObject options, int viewTag, IPromise promise)
+        {
+            //var viewTag = options.Value<int>("view");
             var cameraForView = CameraManager.GetCameraForView(viewTag);
             if (cameraForView == null)
             {
                 promise.Reject("No camera found.");
                 return;
             }
 
-            var mode = options.Value<int>("mode");
-            if (mode == CameraCaptureModeVideo)
-            {
-                if (_recordingTask != null)
-                {
-                    promise.Reject("Cannot run more than one capture operation.");
-                    return;
-                }
+            await CapturePhotoAsync(cameraForView, options, promise).ConfigureAwait(false);
+        }
 
-                _recordingCancellation = new CancellationTokenSource();
-                _recordingTask = RecordAsync(cameraForView, options, promise, _recordingCancellation.Token);
+        [ReactMethod]
+        public void record(JObject options, int viewTag, IPromise promise)
+        {
+            //var viewTag = options.Value<int>("view");
+            var cameraForView = CameraManager.GetCameraForView(viewTag);
+            if (cameraForView == null)
+            {
+                promise.Reject("No camera found.");
```

Comment:
>. [](start = 47, length = 1)

you used ! on the C++ side but . here :) standardize on one

Reply:
Fixed c++ side.